src/data/nat/basic.lean

```diff
@@ -311,6 +323,15 @@ protected theorem div_le_div_right {n m : ℕ} (h : n ≤ m) {k : ℕ} : n / k 
 (nat.eq_zero_or_pos k).elim (λ k0, by simp [k0]) $ λ hk,
 (le_div_iff_mul_le' hk).2 $ le_trans (nat.div_mul_le_self' _ _) h
 
+def le_half_of_double_le (m n : ℕ) (h : 2*m ≤ n) : m ≤ n/2 :=
```

Comment:
This should be a `lemma` instead of a `def`. Also why is this specialized to 2? Is a reason it doesn't generalize to `k`?